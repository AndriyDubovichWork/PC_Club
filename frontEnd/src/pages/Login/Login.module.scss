.login {
  width: 80%;
  height: 80vh;
  margin: 0 auto;

  display: flex;

  flex-direction: column;
  align-items: center;
  justify-content: center;
  .form {
    display: flex;
    flex-direction: column;

    align-items: center;
    .email,
    .password {
      margin: 1vh 0;
    }

    .Link {
      margin-bottom: 1vh;
      color: #000;
      text-decoration: none;
    }
  }
}
