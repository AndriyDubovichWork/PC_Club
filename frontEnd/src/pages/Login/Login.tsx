import React, { useContext } from 'react';
import style from './Login.module.scss';
import { Field, Form, Formik } from 'formik';
import LoginSchema from './Validation/LoginSchema';
import { TextField } from 'formik-mui';
import Button from '../../Styled/Button/Button';
import { Checkbox, FormControlLabel } from '@mui/material';
import { NavLink, useNavigate } from 'react-router-dom';
import { IsAuthorisedContext } from '../../Context/IsAuthorisedContext';

const Login = () => {
  const navigate = useNavigate();

  const initialValues = {
    email: '',
    password: '',
  };
  const { setIsAuthorised } = useContext(IsAuthorisedContext);

  return (
    <main className={style.login}>
      <Formik
        initialValues={initialValues}
        onSubmit={() => {
          setIsAuthorised(true);
          navigate('/PC_Club');
        }}
        validationSchema={LoginSchema}
      >
        <Form className={style.form}>
          <Field
            name='email'
            component={TextField}
            placeholder='Email'
            className={style.email}
          />
          <Field
            name='password'
            component={TextField}
            placeholder='Password'
            className={style.password}
          />
          <FormControlLabel control={<Checkbox />} label='remember me' />
          <NavLink to='/PC_Club/forgotPassword' className={style.Link}>
            Forgot Password?
          </NavLink>
          <Button type='submit'>Log In</Button>
        </Form>
      </Formik>
    </main>
  );
};

export default Login;
