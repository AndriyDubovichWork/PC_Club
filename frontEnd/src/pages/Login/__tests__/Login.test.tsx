import {
  render,
  screen,
  cleanup,
  waitFor,
  fireEvent,
} from '@testing-library/react';
import Login from '../Login';
import renderer from 'react-test-renderer';
import { MemoryRouter, Route, Routes } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

test('Login match snapshot', () => {
  const tree = renderer
    .create(
      <MemoryRouter initialEntries={['/PC_Club/login']}>
        <Routes>
          <Route path='/PC_Club/login' element={<Login />} />
        </Routes>
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('Login contains fields', () => {
  const { getByRole } = render(
    <MemoryRouter initialEntries={['/PC_Club/login']}>
      <Routes>
        <Route path='/PC_Club/login' element={<Login />} />
      </Routes>
    </MemoryRouter>
  );

  expect(getByRole('main')).toBeInTheDocument();

  expect(screen.getByPlaceholderText('Email')).toBeInTheDocument();
  expect(screen.getByPlaceholderText('Password')).toBeInTheDocument();
});
test('Login input cnahges', () => {
  render(
    <MemoryRouter initialEntries={['/PC_Club/login']}>
      <Routes>
        <Route path='/PC_Club/login' element={<Login />} />
      </Routes>
    </MemoryRouter>
  );

  const email: HTMLInputElement = screen.getByPlaceholderText('Email');
  const password: HTMLInputElement = screen.getByPlaceholderText('Password');

  expect(email).toBeInTheDocument();
  expect(password).toBeInTheDocument();

  waitFor(() => {
    fireEvent.change(email, { target: { value: 'invalid' } });
    fireEvent.change(password, { target: { value: 'Rtx 4080' } });
  });

  expect(email.value).toBe('invalid');
  expect(password.value).toBe('Rtx 4080');
});
