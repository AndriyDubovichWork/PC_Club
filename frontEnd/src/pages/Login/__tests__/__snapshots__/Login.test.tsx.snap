// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login match snapshot 1`] = `
<main
  className="login"
>
  <form
    action="#"
    className="form"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root email css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r0:"
          name="email"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Email"
          required={false}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-ihdtdm"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root password css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r1:"
          name="password"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Password"
          required={false}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-ihdtdm"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          className="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate={false}
          onChange={[Function]}
          required={false}
          type="checkbox"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        remember me
      </span>
    </label>
    <a
      className="Link"
      href="/PC_Club/forgotPassword"
      onClick={[Function]}
    >
      Forgot Password?
    </a>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-127p27p-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      Log In
    </button>
  </form>
</main>
`;
